src/Voron/Data/BTrees/TreePage.cs

```diff
@@ -177,7 +189,7 @@ public ushort* KeysOffsets
                 LastMatch = lastMatch;
                 LastSearchPosition = lastSearchPosition;
 
-                if (lastSearchPosition >= numberOfEntries)
+                if (lastSearchPosition<=-1 || lastSearchPosition >= numberOfEntries)
```

Comment:
can we split this into two cases e.g. `if (backward)`

Reply:
Done